File: test-suite/src/test/java/io/micronaut/security/oauth2/docs/openid/OktaAuthenticationMapper.java
package io.micronaut.security.oauth2.docs.openid;

//tag::clazz[]

import io.micronaut.core.annotation.NonNull;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.security.authentication.AuthenticationResponse;
import io.micronaut.security.oauth2.endpoint.authorization.state.State;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdAuthenticationMapper;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdClaims;
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdTokenResponse;
import jakarta.inject.Named;
import jakarta.inject.Singleton;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Flux;

@Singleton
@Named("okta") // <1>
public class OktaAuthenticationMapper implements OpenIdAuthenticationMapper {

    @Override
    @NonNull
    public Publisher<AuthenticationResponse> createAuthenticationResponse(String providerName, // <2>
                                                                          OpenIdTokenResponse tokenResponse, // <3>
                                                                          OpenIdClaims openIdClaims, // <4>
                                                                          @Nullable State state) { // <5>
        return Flux.just(AuthenticationResponse.success("name")); // <6>
    }
}
//end::clazz[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/oauth2/docs/openid/OktaAuthenticationMapper.kt
package io.micronaut.security.oauth2.docs.openid

//tag::clazz[]

import io.micronaut.security.authentication.AuthenticationResponse
import io.micronaut.security.oauth2.endpoint.authorization.state.State
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdAuthenticationMapper
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdClaims
import io.micronaut.security.oauth2.endpoint.token.response.OpenIdTokenResponse
import jakarta.inject.Named
import jakarta.inject.Singleton

@Singleton
@Named("okta") // <1>
class OktaAuthenticationMapper : OpenIdAuthenticationMapper {

    override fun createAuthenticationResponse(providerName: String, // <2>
                                              tokenResponse: OpenIdTokenResponse, // <3>
                                              openIdClaims: OpenIdClaims, // <4>
                                              state: State?) // <5>
            : AuthenticationResponse {
        return AuthenticationResponse.success("name") // <6>
    }
}
//end::clazz[]
